lib/tasks/gitlab/check.rake

```diff
@@ -278,6 +279,29 @@ namespace :gitlab do
       end
     end
 
+    def check_upload_perms
+      print "Upload directory permissions correct? ... "
+
+      upload_path = Rails.root.join(Gitlab.config.gitlab.user_home,"gitlab-rails","uploads")
+      upload_path_tmp = Rails.root.join(upload_path,"tmp")
+
+      # if tmp upload dir has incorrect permissions, assume others do as well
+      if File.stat(upload_path_tmp).mode == 16877 && File.owned?(upload_path_tmp) # verify drwxr-xr-x permissions
```

Comment:
Maybe we should skip this test if upload_path_tmp does not exist. (Does it exist on a fresh install?)

For readability, it might also be nice to write `040755` instead of `16877`.